File: net/ChainOfResponsability/ChainConsole/Chain/ByLong.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChainConsole.Chain
{
    public class ByLong : ChainBase
    {

        public override void Process(string data)
        {
            if (data.Length < 4)
                throw new Exception("To short");

            base.MoveOn(data);
        }
    }
}

File: java/ChainOfResponsability/src/ByLong.java

public class ByLong extends ChainBase {

	@Override
	public void Process(String data) throws Exception {
		
		if (data.length() < 4) {
			System.out.print("Error: By Long");
            throw new Exception();
		}

        super.MoveOn(data);
	}

}